docs/concepts/workloads/controllers/garbage-collection.md

```diff
@@ -63,33 +63,50 @@ metadata:
   ...
 ```
 
-## Controlling whether and how the garbage collector deletes dependents
-
-When you delete object, you can specify whether the object's dependents are
-deleted automatically. If you delete an object without deleting its dependents
-automatically, the dependents are said to be *orphaned*. Deleting dependents
-automatically is called *cascading deletion*. Further, there are two modes of
-*cascading deletion*: if the object is deleted immediately and the garbage
-collector then deletes the dependents in the background, it is called
-*background cascading deletion*. In contrast, in *foreground cascading
-deletion*, the object first enters a "deletion in progress" state, where the
-object is still visible via the REST API, its `deletionTimestamp` is set, and
-its metadata.finalizers contains "foregroundDeletion". Then the garbage
-collector deletes the dependents. Once the garbage collector has deleted all
-dependents whose ownerReference.blockOwnerDeletion=true, it will finally delete
-the object.
+## Controlling how the garbage collector deletes dependents
+
+When you delete an object, you can specify whether the object's dependents are
+also deleted automatically. Deleting dependents automatically is called *cascading
+deletion*.  There are two modes of *cascading deletion*: *background* and *foreground*. 
+
+If you delete an object without deleting its dependents
+automatically, the dependents are said to be *orphaned*. 
+
+### Background cascading deletion
+
+In *background cascading deletion*, Kubernetes deletes the owner object 
+immediately and the garbage collector then deletes the dependents in 
+the background.
+
+### Foreground cascading deletion
+
+In *foreground cascading deletion*, the root object first
+enters a "deletion in progress" state. In the "deletion in progress" state,
+the following things are true:
+
+ * The object is still visible via the REST API
+ * The object's `deletionTimestamp` is set
+ * The object's `metadata.finalizers` contains the value "foregroundDeletion".
+ 
+Once the "deletion in progress" state is set, the the garbage
+collector deletes the object's dependents. Once the garbage collector has deleted all
+dependents (objects with `ownerReference.blockOwnerDeletion=true`), it delete
```

Comment:
Dependents with `ownerReference.blockOwnerDeletion=true` is a special kind of dependents, so how about

```
Once the garbage collector has deleted all "blocking" dependents (dependents with ownerReference.blockOwnerDeletion=true`), ...
```